src/Shared/EmptyListWrapper.js

```diff
@@ -0,0 +1,169 @@
+import React, { Component } from 'react'
+import { View, Text, StyleSheet, Image } from 'react-native'
+import Icon from 'react-native-vector-icons/dist/MaterialIcons'
+
+import WrappedTextButton from '../TextButton/TextButton.js'
+
+export default class EmptyListWrapper extends Component {
+  render() {
+    let { listEmptyState, items, children } = this.props
+
+    if (!listEmptyState) {
+      return <View>Loading...</View>
+    }
+
+    let {
+      type,
+      text,
+      icon,
+      primaryColor,
+      contrastColor,
+      borderRadius,
+      shadow,
+      upperCase,
+      action,
+      buttonWidth,
+    } = listEmptyState
+
+    if (!items || items.length === 0) {
+      return (
+        <>
+          <ImageHolder listEmptyState={listEmptyState} />
+          {type !== 'noButton' && (
+            <WrappedTextButton
+              type={type}
+              text={text}
+              icon={icon}
+              primaryColor={primaryColor}
+              contrastColor={contrastColor}
+              borderRadius={borderRadius}
+              shadow={shadow}
+              upperCase={upperCase}
+              action={action}
+              container={{
+                width: buttonWidth,
+                alignSelf: 'center',
+              }}
+            />
+          )}
+        </>
+      )
+    } else {
+      return <>{children}</>
+    }
+  }
+}
+
+class ImageHolder extends Component {
```

Comment:
Because these classes are so small and just have a render function, I think they would be great candidates to convert to functional components. It'll make things a little more elegant.